internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -140,12 +144,66 @@ func (h *Host) SetConfidentialUVMOptions(ctx context.Context, r *guestresource.L
 	return nil
 }
 
-// InjectFragment extends current security policy with additional constraints
-// from the incoming fragment.
-//
-// TODO (maksiman): add fragment validation and injection logic
-func (*Host) InjectFragment(ctx context.Context, fragment *guestresource.LCOWSecurityPolicyFragment) (err error) {
-	log.G(ctx).WithField("fragment", fragment).Debug("fragment received in guest")
+/*
+   InjectFragment extends current security policy with additional constraints
+   from the incoming fragment. Note that it is base64 encoded over the bridge/
+
+   There are three checking steps:
+
+	1 - Unpack the cose document and check it was actually signed with the cert chain inside its header
+	2 - Check that the issuer field did:x509 identifier is for that cert chain (ie fingerprint of a non leaf cert
+	    and the subject matches the leaf cert)
+	3 - Check that this issuer/feed match the requirement of the user provided security policy (done in the rego
+	    by LoadFragment)
+*/
+
+func (h *Host) InjectFragment(ctx context.Context, fragment *guestresource.LCOWSecurityPolicyFragment) (err error) {
+	log.G(ctx).WithField("fragment", fmt.Sprintf("%+v", fragment)).Debug("GCS Host.InjectFragment")
+
+	raw, err := base64.StdEncoding.DecodeString(fragment.Fragment)
+	if err != nil {
+		return err
+	}
+	blob := []byte(fragment.Fragment)
+	// keep a gopy of the fragment so we can manually figure out what went wrong
+	// will be removed eventually.
+	_ = os.WriteFile("/tmp/fragment.blob", blob, 0644)
+
+	unpacked, err := cosesign1.UnpackAndValidateCOSE1CertChain(raw, nil, nil, true)
+
```

Comment:
extra space

Reply:
Do you mean line? Which one? And why exactly?

Reply:
its idiomatic to keep the error definition and `if` clause together, so:

```go
unpacked, err := cosesign1.UnpackAndValidateCOSE1CertChain(raw, nil, true)
if err != nil {
    return ...
}
```

(at least, ive been told to avoid putting spaces between them during PRs...)

Reply:
ok